models/image.py

```diff
@@ -1741,139 +1721,186 @@ def query_images(
+                           "                                       i.ra_corner_11, i.dec_corner_11, "
+                           "                                       i.ra_corner_10, i.dec_corner_10 ]) " )
+                    subdict[ 'ra' ] = ra
+                    subdict[ 'dec' ] = dec
+                    andtxt = " AND "
+
+            # A few fields need preprocessing before feeding into the code below
+            min_dateobs = None if min_dateobs is None else parse_dateobs(min_dateobs, output='mjd')
+            max_dateobs = None if max_dateobs is None else parse_dateobs(max_dateobs, output='mjd')
+            types = None if type is None else [ ImageTypeConverter.to_int(t) for t in listify(type) ]
+
+            fields = [ { 'field': 'project',             'val': project,        'type': 'list' },
+                       { 'field': 'target',              'val': target,         'type': 'list' },
+                       { 'field': 'section_id',          'val': section_id,     'type': 'list' },
+                       { 'field': 'filter',              'val': filter,         'type': 'list' },
+                       { 'field': 'instrument',          'val': instrument,     'type': 'list' },
+                       { 'field': 'provenance_id',       'val': provenance_ids, 'type': 'list' },
+                       { 'field': '_type',               'val': types,          'type': 'list' },
+                       { 'field': 'mjd',                 'val': min_mjd,        'type': 'ge' },
+                       { 'field': 'mjd',                 'val': min_dateobs,    'type': 'ge' },
+                       { 'field': 'mjd',                 'val': max_mjd,        'type': 'le' },
+                       { 'field': 'mjd',                 'val': max_dateobs,    'type': 'le' },
+                       { 'field': 'exp_time',            'val': min_exp_time,   'type': 'ge' },
+                       { 'field': 'exp_time',            'val': max_exp_time,   'type': 'le' },
+                       { 'field': 'fwhm_estimate',       'val': min_seeing,     'type': 'ge' },
+                       { 'field': 'fwhm_estimate',       'val': max_seeing,     'type': 'le' },
+                       { 'field': 'lim_mag_estimate',    'val': min_lim_mag,    'type': 'ge' },
+                       { 'field': 'lim_mag_estimate',    'val': max_lim_mag,    'type': 'le' },
+                       { 'field': 'airmass',             'val': min_airmass,    'type': 'ge' },
+                       { 'field': 'airmass',             'val': max_airmass,    'type': 'le' },
+                       { 'field': 'zero_point_estimate', 'val': min_zero_point, 'type': 'ge' },
+                       { 'field': 'zero_point_estimate', 'val': max_zero_point, 'type': 'le' },
+                       { 'field': 'bkg_rms_estimate',    'val': min_background, 'type': 'ge' },
+                       { 'field': 'bkg_rms_estimate',    'val': max_background, 'type': 'le' } ]
```

Comment:
Yeah.  I completely refactored this when I got rid of query_images.  query_images was a nice idea -- it produced a partial SQLAlchemy query that you could add further things to.  However, it wasn't used anywhere other than the tests.  And, to implement the overlap search criterion, I really wanted to use temp tables, which wasn't really consistent with the query_images way of doing things.  When I was looking at it, I realized that there were a whole lot of little if blocks for each and every one of these fields, and that they could all be unified into a single block of code with just a few if/thens and a table like this.

Incidentally, I think the linter I complained about above (black) would remove all the extra spaces here, because it doesn't match the official style, and that would make the whole thing less readable....